packages/core/echo/echo-schema/src/query/query.ts

```diff
@@ -149,8 +149,12 @@ export class Query<T extends TypedObject = TypedObject> {
 
   private _ensureCachePresent() {
     if (!this._resultCache) {
-      this._resultCache = Array.from(this._sources).flatMap((source) => source.getResults()) as QueryResult<T>[];
-      this._objectCache = this._resultCache.map((result) => result.object!).filter((object): object is T => !!object);
+      this._resultCache = new EchoArray(
```

Comment:
This doesnt look right. The idea behind EchoArray is to be a data-access object: the changes made to EchoArray are automatically serialized as mutations and persisted on disk. 

Is there a way to instead make `flatMap` work with `EchoArray` natively?
e.g. `[new EchoArray([1, 2]), new EchoArray([3])].flatMap(x => x) == [1, 2, 3]`

Reply:
yeah, this didn't feel right, considered creating a flatMap decorator instead of it

but it's nice that Proxying to an empty array instance worked. tests seem to pass, but I'm still a bit scared that this might not be the most robust construct and smth can break along the way. but I guess we'll fix it when it comes to it